aiohttp/resolver.py

```diff
@@ -6,16 +6,19 @@
 
 __all__ = ("ThreadedResolver", "AsyncResolver", "DefaultResolver")
 
-
 try:
     import aiodns
-
-    aiodns_default = hasattr(aiodns.DNSResolver, "getaddrinfo")
-except ImportError:
+    import pycares
+
+    # Only use aiodns if it has the getaddrinfo method and the
+    # underlying pycares library can locate DNS servers.
+    aiodns_default = hasattr(aiodns.DNSResolver, "getaddrinfo") and bool(
+        pycares.Channel().servers
+    )
+except Exception:  # may raise ImportError or AresError
```

Comment:
I don't really like the broad exception catch here.. but it was far more complex than trying to handle AresError for a library we don't know is installed or not